.github/workflows/pre_commit.yml

```diff
@@ -31,3 +31,19 @@ jobs:
         git config --global diff.wsErrorHighlight "all"
     - uses: actions/setup-python@v3.0.0
     - uses: pre-commit/action@v2.0.3
+    - name: Check changelog entries
```

Comment:
Don't add this check to Github workflows. This way `pre-commit` is not useful when installed locally. You can add it properly to `pre-commit`. `pre-commit` is very extensible, with scripts, regex and other languages. See https://pre-commit.com/index.html#new-hooks . The best portable way is to write Python scripting if what you want to achieve is not possible with e.g. `fail` hook type, with matching regex files. You have an example of scripts from the community for a similar thing you want to do, see https://github.com/Sceptre/sceptrelint .

Reply:
The reason to use python scripting instead of bash, is that Python is a core dependency of pre-commit, so you don't rely on the system requirement of having, e.g. `bash`.

Reply:
pre-commit looks like a deprecated action to me.

Reply:
@ljmf00 : The way @ibuclaw implemented it makes sense to me, as it achieve the goal of preventing PR from being merged if the changelog would be ignored. Having the ability to have the users run the same check before they commit / push is a good improvement, but it's secondary, so I suggest we move ahead with the merge and anyone willing to improve it is more than welcome to.

Reply:
> pre-commit looks like a deprecated action to me.

It's a complicated story. Basically, it is deprecated because there's a `pre-commit` Github application and the dev considers it deprecated as it can lead to insecure usage (if used with the push permission when auto-fix is enabled). There are other reasons to be deprecated, as `pre-commit` is backed by a company (PRE-COMMIT CI, LLC) and there might be interest in not maintaining the action. We don't use push permissions to auto-fix changes, so it is fine to use it.

The community wants to maintain the action, but the maintainer is a bit arrogant, so suggesting upstream changes lead to rejection, issue blocking, or even user blocking (Several issues demonstrating that https://github.com/pre-commit/pre-commit/issues/1982#issuecomment-894792831 , https://github.com/pre-commit/action/issues/148). It is kind of unfortunate, but I'm blocked by him due to forking his project, see the last referenced issue. I'm maintaining a work-in-progress fork of the action (https://github.com/before-commit/run-action), but currently, the action we are using is going to indefinitely work with updated `pre-commit`, as it uses `pip` to download the package. I never switched to the forked action as I don't have enough time to make it stable with all the features I want to add.

Reply:
> @ljmf00 : The way @ibuclaw implemented it makes sense to me, as it achieve the goal of preventing PR from being merged if the changelog would be ignored. Having the ability to have the users run the same check before they commit / push is a good improvement, but it's secondary, so I suggest we move ahead with the merge and anyone willing to improve it is more than welcome to.

This is just a rudimentary test to prevent people from checking in changelogs that won't get picked up.

It could do more of course - such as check the git commit message, and complain that a PR _lacks_ a changelog entry if (1) any compiler/runtime files have changed and (2) top message doesn't contain:
- "fix Issue "
- "refactor: "
- etc...

Reply:
Mainline already has two non-compliant changelogs. :shrug: 
https://github.com/dlang/dmd/tree/master/changelog